magma/compile_guard.py

```diff
@@ -17,6 +19,38 @@
 
 _logger = root_logger().getChild("compile_guard")
 
+_compile_guard_builder_stack = Stack()
+
+
+def _get_compile_guard_builder_stack() -> Stack:
+    global _compile_guard_builder_stack
+    return _compile_guard_builder_stack
+
+
+@contextlib.contextmanager
+def _push_compile_guard_builder_stack(builder: '_CompileGuardBuilder'):
+    _get_compile_guard_builder_stack().push(builder)
+    yield
+    _get_compile_guard_builder_stack().pop()
+
+
+def get_active_compile_guard_info() -> Iterable['CompileGuardInfo']:
+    stack = _get_compile_guard_builder_stack()
+    offset = 0
+    while True:
+        try:
+            builder = stack.peek(offset=offset)
+        except IndexError:
+            break
+        yield builder.info
+        offset += 1
```

Comment:
Perhaps we could add an `Iterable` interface to the stack and use it here?

Reply:
:+1: